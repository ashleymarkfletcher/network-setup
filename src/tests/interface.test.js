import React from 'react';
import renderer from 'react-test-renderer';
import Interface from '../components/interface';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';

test('Interface renders', () => {

  const mockConfig = {
    errors: {},
    gateway: "192.168.1.1",
    id: "_8h7ez2224",
    ip: "192.168.1.10",
    primaryDNS: "8.8.8.8",
    secondaryDNS: "8.8.4.4",
    subnet: "255.255.255.0"
  }

  const configureInterface = () => {}

  const deleteConfig = () => {}

  const save = () => {}

  const tree = shallow(
    <Interface
      config={mockConfig}
      save={save}
      deleteConfig={deleteConfig}
      configureInterface={configureInterface}
      key={mockConfig.id}
    >
    </Interface>
  )

  expect(shallowToJson(tree)).toMatchSnapshot();
})
