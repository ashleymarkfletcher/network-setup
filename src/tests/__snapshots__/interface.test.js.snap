// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Interface renders 1`] = `
<Paper
  circle={false}
  rounded={true}
  style={
    Object {
      "margin": 20,
      "padding": 20,
    }
  }
  transitionEnabled={true}
  zDepth={1}
>
  <div
    className="interface"
  >
    <TextField
      disabled={false}
      errorText=""
      floatingLabelFixed={false}
      floatingLabelText="IP address"
      fullWidth={false}
      hintText="192.168.x.x"
      multiLine={false}
      name="ip"
      onChange={[Function]}
      rows={1}
      style={
        Object {
          "marginRight": 10,
          "width": 125,
        }
      }
      type="text"
      underlineShow={true}
      value="192.168.1.10"
    />
    <TextField
      disabled={false}
      errorText=""
      floatingLabelFixed={false}
      floatingLabelText="Subnet Mask"
      fullWidth={false}
      hintText="255.255.x.x"
      multiLine={false}
      name="subnet"
      onChange={[Function]}
      rows={1}
      style={
        Object {
          "marginRight": 10,
          "width": 125,
        }
      }
      type="text"
      underlineShow={true}
      value="255.255.255.0"
    />
    <TextField
      disabled={false}
      errorText=""
      floatingLabelFixed={false}
      floatingLabelText="Gateway"
      fullWidth={false}
      hintText="192.168.x.x"
      multiLine={false}
      name="gateway"
      onChange={[Function]}
      rows={1}
      style={
        Object {
          "marginRight": 10,
          "width": 125,
        }
      }
      type="text"
      underlineShow={true}
      value="192.168.1.1"
    />
    <TextField
      disabled={false}
      errorText=""
      floatingLabelFixed={false}
      floatingLabelText="Primary DNS"
      fullWidth={false}
      hintText="192.168.x.x"
      multiLine={false}
      name="primaryDNS"
      onChange={[Function]}
      rows={1}
      style={
        Object {
          "marginRight": 10,
          "width": 125,
        }
      }
      type="text"
      underlineShow={true}
      value="8.8.8.8"
    />
    <TextField
      disabled={false}
      errorText=""
      floatingLabelFixed={false}
      floatingLabelText="Secondary DNS"
      fullWidth={false}
      hintText="192.168.x.x"
      multiLine={false}
      name="secondaryDNS"
      onChange={[Function]}
      rows={1}
      style={
        Object {
          "marginRight": 10,
          "width": 125,
        }
      }
      type="text"
      underlineShow={true}
      value="8.8.4.4"
    />
    <RaisedButton
      backgroundColor="#4CAF50"
      disabled={false}
      fullWidth={false}
      icon={
        <ContentSave
          color="#FFFFFF"
        />
      }
      labelPosition="after"
      onClick={[Function]}
      primary={false}
      secondary={false}
      style={
        Object {
          "marginLeft": 10,
        }
      }
    />
    <RaisedButton
      backgroundColor="#F44336"
      disabled={false}
      fullWidth={false}
      icon={
        <ActionDelete
          color="#FFFFFF"
        />
      }
      labelPosition="after"
      onClick={[Function]}
      primary={false}
      secondary={false}
      style={
        Object {
          "marginLeft": 10,
        }
      }
    />
    <RaisedButton
      backgroundColor="#03A9F4"
      disabled={false}
      fullWidth={false}
      icon={
        <ContentSend
          color="#FFFFFF"
        />
      }
      label="Configure"
      labelColor="#FFFFFF"
      labelPosition="after"
      onClick={[Function]}
      primary={false}
      secondary={false}
      style={
        Object {
          "marginLeft": 10,
        }
      }
    />
  </div>
</Paper>
`;
